docs/design/storagenode-graceful-exit.md

```diff
@@ -0,0 +1,88 @@
+# Storage Node Graceful Exit Design Document
+
+
+## Overview
+
+When a Storage Node wants to leave the network but does not want to lose their escrow we need to have a mechanism for them to exit the network “gracefully”.
+
+This process including the Storage nodes transferring their pieces to other nodes so that the satellite does not have to repair those pieces because of a node exiting abruptly. The process of a Storage Node exiting gracefully including that node requesting a list of Storage Nodes to send their pieces to and updating the satellite with what nodes are now storing those pieces. Graceful Exit for Storage Nodes is beneficially for both the Storage Node and satellite because the storage node receives their escrow and the satellite saves money from not having to repair files.
+
+
+## Goals
+
+- Give Storage Nodes a mechanism to leave the network while receiving their escrows, and reduce repair caused by node churn on satellites. 
+
+
+## Non Goals
+
+- Sending the storage nodes escrows.
+	- The sending of tokens will happen through our normal token payment process.
+
+
+## Scenarios (MVP)
+
+- A storage node gracefully exits a satellite or all satellites.
+- A storage node fails to complete the graceful exit process.
+
+
+## Scenarios (Non MVP)
+
+- A storage node runs out of bandwidth during the graceful exit process.
+- A storage node wants to “partially” exit a satellite.
+- A storage node wants to gracefully exit one satellite.
+- A storage node wants to rejoin a satellite it previously exited.
+
+## Business Requirements/ Job Stories (MVP)
+
+- When a Storage Node runs the graceful exit command I want them to be prompted with a confirmation message so that we can avoid storage nodes running the command accidentally.
+- When a Storage Node gracefully exits the network I want all of the pieces they have been storing to be deleted from their hard drive so that their computers hard drive space is no longer used. 
+- When a Storage node triggers graceful exit I want them to be omitted from the node selection process for uploads so they are no longer selected to store new pieces.
+- When a Storage node triggers graceful exit I want thier allocated bandwidth on the network to be ignored so that they can complete their graceful exit as quickly as possible.
+- When a node dose not complete the graceful exit entirely I want the satellite to keep track of this so that they are subject to be DQed and their escrow kept
+	- this includes the node sending bad data to other nodes
+	- the node not transferring some pieces it holds
+- When a Storage Node leaves the network I want the ability to run a report on the satellite to get information about what Storage Nodes exited so that I can pay them thier escrow amounts. 
+	- Create a report on the satellite to display which nodes have exited gracefully during a specified timeframe. The report must include:
```

Comment:
I'm not sure if this is a typo or it's ok, according to my speller it's `timeframe` => `time frame`